packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1280,51 +1282,71 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
         throw new Error('Setting \'instanceType\' is required when \'launchTemplate\' and \'mixedInstancesPolicy\' is not set');
       }
 
+      Tags.of(this).add(NAME_TAG, this.node.path);
```

Comment:
I'd like for someone to take a second look at this. Should we only be adding this tag as part of a launch configuration? The PR this was added in mentions that this was added to support tagging of AutoScalingGroups and Security Groups so I'm thinking this isn't specific to launch configurations or launch templates, but AutoScalingGroups as a whole.